k-factor.sh

```diff
@@ -91,8 +128,9 @@ G1 X20 Y0 E${EXT_20} F${SLOW_SPEED}
 G1 E-${RETRACTION}
 G1 X-80 Y5 F${MOVE_SPEED}
 EOF
-	i=$(($i+$K_STEPPING))
-	done
+  (( i+=1 ))
```

Comment:
Not sure why my input did get lost...
Here there are again:
bash procedure can only work integer... If K_STEPPING=7.5, then bash will complain... using what I propose is a solution against that problem: I count the number of iterations, start from zero then increment (by 1) till I get the count... This produce exactly the same result but allows for NON INTEGER K_STEPPING...

Reply:
It is about this line:
`M900 K$((i)) ; set K-factor`
You will get the right number of steps for float K-Factors but the wrong gcode. You could probably go by 
`M900 K$((i * K_STEPPING + K_START)) ; set K-factor`

Reply:
Actually it would need:
```
cnt=$(awk -v krange="$K_RANGE" -v kstep="$K_STEPPING" 'BEGIN {print krange/kstep}')
i=0
while [ ${i} -le ${cnt} ]; do
	cat << EOF
M900 K$(awk -v kstart="$K_START" -v kstep="$K_STEPPING" -v num="$i" 'BEGIN {print kstep * num + kstart}') ; set K-factor
G1 E${RETRACTION}
G1 X20 Y0 E${EXT_20} F${SLOW_SPEED}
G1 X40 Y0 E${EXT_40} F${FAST_SPEED}
G1 X20 Y0 E${EXT_20} F${SLOW_SPEED}
G1 E-${RETRACTION}
G1 X-80 Y5 F${MOVE_SPEED}
EOF
	i=$((i+1))
	done
```
This makes the loop construct easier to read than my current solution.